extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/BuildTimeEnabledProcessor.java

```diff
@@ -271,9 +271,9 @@ private void transformBean(AnnotationTarget target, TransformationContext ctx, b
             } else {
                 // Add @Alternative to the producer
                 transform.add(DotNames.ALTERNATIVE);
+                transform.remove(s -> s.name().equals(DotNames.ALTERNATIVE_PRIORITY));
```

Comment:
This solution is not ideal, i.e. if a user declares an `@Alternative` producer and enables it via `quarkus.arc.selected-alternatives` then `@UnlessBuildProperty` would have no effect. OTOH it's definitely better than the current solution and we have to think a little how to solve this properly. So +1 for this PR and I'll create a separate issue.